workflows_core/dataset/dataset.py

```diff
@@ -48,14 +49,17 @@ def delete(self):
 
     def insert_documents(
         self,
-        documents: List[document.Document],
+        documents: Union[List[document.Document], DocumentList],
         use_json_encoder: bool = True,
         *args,
         **kwargs
     ) -> Dict[str, Any]:
-        for index, document in enumerate(documents):
-            if hasattr(document, "to_dict"):
-                documents[index] = document.to_dict()
+        if hasattr(documents, "to_json"):
+            documents = documents.to_json()
+        else:
+            for index in range(len(documents)):
+                if hasattr(documents[index], "to_json"):
+                    documents[index] = documents[index].to_json()
         if use_json_encoder:
```

Comment:
I think there is some redundancy here. You don't need to use JSON encoder here you are already converting it to `to_json`